collated/test/A0127014W.md

```diff
@@ -0,0 +1,258 @@
+        String date = "thirteenth september";
+        List<Date> dates = CommandHelper.convertStringToMultipleDates(date);
+        assertTrue(dates.get(0).getDate() == 13);
+        assertTrue(dates.get(0).getMonth() == 8);
+    }
+
+    @Test
+    public void convertStringToMultipleDates_correct_dayOfWeek_timeOfDayInNumbers() {
+        String date = "friday 7pm";
+        List<Date> dates = CommandHelper.convertStringToMultipleDates(date);
+        assertTrue(dates.get(0).getDay() == 5);
+        assertTrue(dates.get(0).getHours() == 19);
+
+        String date2 = "next wed 3 afternoon";
+        List<Date> dates2 = CommandHelper.convertStringToMultipleDates(date2);
+        assertTrue(dates2.get(0).getDay() == 3);
+        assertTrue(dates2.get(0).getHours() == 15);
+
+        String date3 = "this monday 1600 hours";
+        List<Date> dates3 = CommandHelper.convertStringToMultipleDates(date3);
+        assertTrue(dates3.get(0).getDay() == 1);
+        assertTrue(dates3.get(0).getHours() == 16);
+    }
+
+    @Test
+    public void convertStringToMultipleDates_timeOfDayInWords() {
+        String date = "next friday seven evening";
+        List<Date> dates = CommandHelper.convertStringToMultipleDates(date);
+        assertTrue(dates.get(0).getDay() == 5);
+        assertTrue(dates.get(0).getHours() == 19);
+
+        String date2 = "3rd april five in the morning";
+        List<Date> dates2 = CommandHelper.convertStringToMultipleDates(date2);
+        assertTrue(dates2.get(0).getDate() == 3);
+        assertTrue(dates2.get(0).getHours() == 5);
+
+        String date3 = "6 april three pm";
+        List<Date> dates3 = CommandHelper.convertStringToMultipleDates(date3);
+        assertTrue(dates3.get(0).getDate() == 6);
+        assertTrue(dates3.get(0).getHours() == 15);
+    }
+
+    @Test
+    public void convertStringToMultipleDates_yearWithTime() {
+        String date = "6 april 2004 3 pm";
+        List<Date> dates = CommandHelper.convertStringToMultipleDates(date);
+        assertTrue(dates.get(0).getDate() == 6);
+        assertTrue(dates.get(0).getHours() == 15);
+        assertTrue(dates.get(0).getYear() == 2004 - 1900);
+    }
+
+    @Test
+    public void convertStringToMultipleDates_getsMultipleDates() {
+        String date = "20 aug and 17 april";
+        List<Date> dates = CommandHelper.convertStringToMultipleDates(date);
+        assertTrue(dates.size() == 2);
+    }
+
+    @Test
+    public void convertStringToDate_shortName_correct_date_month() {
+        String dateString = "13 sep";
+        Date date;
+        try {
+            date = CommandHelper.convertStringToDate(dateString);
+            assertTrue(date.getDate() == 13);
+            assertTrue(date.getMonth() == 8);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void convertStringToDate_multipleDates_exceptionThrown(){
+        String dateString = "13 sep and 14 oct";
```

Comment:
there is a better way to expect exceptions in junit tests. please check how and make changes appropriately.